app/data/index.js

```diff
@@ -202,6 +202,45 @@ module.exports = Base => class extends Base {
         });
     }
 
+    readPlayerWelcome(playerName) {
+        //this applies to homes and shared homes
+        return new Promise((resolve) => {
+            Player.findOne({ name: playerName }, (err, player) => {
+                if (err) return reject(err);
+                resolve(player.welcome);            
+            });
+        });
+    }
+
+    updatePlayerWelcome(playerName, input) {
```

Comment:
Too much logic in this. most all of this logic should live outside of this function.. Most anything in the data file should simply take what you've already worked out and add it